operator/src/intTest/java/org/projectnessie/operator/testinfra/K3sContainerLifecycleManager.java

```diff
@@ -0,0 +1,518 @@
+        name: otel-collector
+        namespace: kube-system
+      spec:
+        repo: https://open-telemetry.github.io/opentelemetry-helm-charts
+        chart: opentelemetry-collector
+        targetNamespace: otel-collector
+        createNamespace: true
+        valuesContent: |-
+          mode: deployment
+          image:
+            repository: "otel/opentelemetry-collector-k8s"
+          ports:
+           jaeger-compact:
+             enabled: false
+           jaeger-thrift:
+             enabled: false
+           jaeger-grpc:
+             enabled: false
+           zipkin:
+             enabled: false
+          config:
+           receivers:
+             jaeger: null
+             prometheus: null
+             zipkin: null
+           exporters:
+             debug:
+               verbosity: detailed
+               sampling_initial: 1
+               sampling_thereafter: 1
+           service:
+             pipelines:
+               traces:
+                 exporters:
+                   - debug
+                 receivers:
+                   - otlp
+               logs: null
+               metrics: null
+      """;
+
+  @Language("Shell Script")
+  private static final String IMAGE_IMPORT_SCRIPT =
+      """
+      #!/usr/bin/env bash
+      set -e
+      TOOL="$(which docker > /dev/null && echo docker || echo podman)"
+      ${TOOL} image save projectnessie/nessie-test-server:$NESSIE_VERSION | \
+        ${TOOL} exec --interactive $CONTAINER_NAME ctr images import --no-unpack -
+      """;
+
+  @Target(ElementType.FIELD)
+  @Retention(RetentionPolicy.RUNTIME)
+  public @interface NessieUri {}
+
+  @Target(ElementType.FIELD)
+  @Retention(RetentionPolicy.RUNTIME)
+  public @interface PrometheusUri {}
+
+  private URI nessieUri;
+  private URI prometheusUri;
+
+  private OpenShiftClient k8sClient;
+
+  private boolean monitoring;
+  private boolean ingress;
+  private boolean telemetry;
+  private boolean waitForComponents;
+
+  @Override
+  public void init(Map<String, String> initArgs) {
+    monitoring = Boolean.parseBoolean(initArgs.getOrDefault("monitoring", "false"));
+    ingress = Boolean.parseBoolean(initArgs.getOrDefault("ingress", "false"));
+    telemetry = Boolean.parseBoolean(initArgs.getOrDefault("telemetry", "false"));
+    waitForComponents = Boolean.parseBoolean(initArgs.getOrDefault("waitForComponents", "false"));
+  }
+
+  @Override
+  protected K3sContainer createContainer() {
+    K3sContainer container =
+        new K3sContainer(dockerImage("k3s").asCompatibleSubstituteFor("rancher/k3s"));
+    List<Integer> exposedPorts = new ArrayList<>();
+    List<String> commandParts = new ArrayList<>();
+    commandParts.add("server");
+    commandParts.add("--tls-san=" + container.getHost());
+    // Mitigate eviction issues in CI by setting eviction thresholds for nodefs very low
+    commandParts.add("--kubelet-arg=eviction-hard=nodefs.available<1%,nodefs.inodesFree<1%");
+    if (monitoring) {
```

Comment:
```suggestion
    commandParts.add("--kubelet-arg=feature-gates=KubeletInUserNamespace=true");
    if (monitoring) {
```

This _seems_ to be necessary for rootless containers. Still not there yet. So many resources and configs :disappointed: